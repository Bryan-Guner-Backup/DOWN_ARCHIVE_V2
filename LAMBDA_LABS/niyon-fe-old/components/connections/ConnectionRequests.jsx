import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Link from 'next/link';
import { Heading2 } from '../~common';

const Wrapper = styled.div`
    box-sizing: border-box;
    width: 100%;
    border-radius: 8px;
    margin: 1rem 0;
    display: flex;
    flex-direction: column;
    transition: max-height 100ms;
`;

const Header = styled.div`
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
`;

const CardWrapper = styled.div`
    box-sizing: border-box;
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;
    align-items: flex-end;
    width: 100%;
`;

const Person = styled.div`
  box-sizing: border-box;
  width: auto;
  height: 60px;
  border-radius: 40px;
  background: #eaeaea;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 8px 6px;
  padding-right: 32px;
  margin: 8px 0;
  margin-right: 32px;
  transition: transform 400ms;

  .img-wrapper {
    width: 52px;
    height: 52px;
    border-radius: 50%;
    overflow: hidden;
    margin-right: 8px;
  }

  .content {
    display: flex;
    flex-direction: column;
  }

  small {
    color: #484848;
  }

  p {
    margin: 5px 0;
  }

  &:hover {
    transform: scale(1.05);
    cursor: pointer;
  }
`;

function ConnectionRequests({ requests, connections }) {
  console.log(requests);

  return (
    <Wrapper>
      <Header>
        <Heading2>Connection Requests</Heading2>
      </Header>
      <CardWrapper>
        { !requests.length ? <div>You have no connection requests</div> : requests.map(({ connection }) => (
          <Link href={{
            pathname: '/profile',
            query: { id: connection.id }
          }}
          >
            <Person>
              <div className="img-wrapper">
                <img src={connection.profile_picture || 'https://image.flaticon.com/icons/svg/660/660611.svg'} alt="" height="100%" />
              </div>
              <div className="content">
                <small>{connection.username}</small>
                <p>{connection.first_name} {connection.last_name}</p>
              </div>
            </Person>
          </Link>
        )) }
      </CardWrapper>
    </Wrapper>
  );
}

ConnectionRequests.propTypes = {
  requests: PropTypes.arrayOf().isRequired,
  connections: PropTypes.arrayOf().isRequired
};

export default ConnectionRequests;
