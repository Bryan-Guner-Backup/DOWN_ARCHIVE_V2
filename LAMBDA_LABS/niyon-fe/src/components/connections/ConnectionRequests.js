import React from 'react'
import Paper from '@material-ui/core/Paper'
import SwipeTabsRequests from './SwipeTabsRequests'

function ConnectionRequests (props) {
  return (

    <div className="connection-req" data-test="connections">
      <div data-test="second-wrapper">
        <Paper className="paper" elevation={1}>
          <h1 className="container-header" data-test="my-connections">
          Connection Requests ({props.sumRequests})
          </h1>
          <SwipeTabsRequests/>
        </Paper>
      </div>
    </div>
  )
}

export default ConnectionRequests
