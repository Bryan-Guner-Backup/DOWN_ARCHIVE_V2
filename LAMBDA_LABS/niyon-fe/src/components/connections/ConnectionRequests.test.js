import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import ConnectionRequests from './ConnectionRequests'

describe('<ConnectionRequests /> component', () => {
  let component
  beforeEach(() => {
    component = shallow(<ConnectionRequests />)
  })

  it('should match the snapshot test', () => {
    expect(toJSON(component)).toMatchSnapshot()
  })

  it('should render the component correctly', () => {
    expect(component.exists()).toBe(true)
  })
})
