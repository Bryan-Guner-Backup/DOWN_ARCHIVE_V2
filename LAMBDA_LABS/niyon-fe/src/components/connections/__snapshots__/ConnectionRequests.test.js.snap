// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConnectionRequests /> component should match the snapshot test 1`] = `
<div
  className="connection-req"
  data-test="connections"
>
  <div
    data-test="second-wrapper"
  >
    <WithStyles(ForwardRef(Paper))
      className="paper"
      elevation={1}
    >
      <h1
        className="container-header"
        data-test="my-connections"
      >
        Connection Requests (
        )
      </h1>
      <SwipeTabsRequests />
    </WithStyles(ForwardRef(Paper))>
  </div>
</div>
`;
